src/utils/analytics.ts

```diff
@@ -0,0 +1,62 @@
+import { getSetting } from "~settings";
+import { ExtensionStorage, TempTransactionStorage } from "./storage";
+import { AnalyticsBrowser } from "@segment/analytics-next";
+
+const analytics = AnalyticsBrowser.load({
+  writeKey: process.env.PLASMO_PUBLIC_SEGMENT_WRITEKEY
+});
+
+export enum EventType {
+  FUNDED = "FUNDED",
+  CONNECTED_APP = "CONNECTED_APP",
+  LOGIN = "LOGIN",
+  ONBOARDED = "ONBOARDED"
+}
+
+export const trackEvent = async (eventName: EventType, properties: any) => {
+  // first we check if we are allowed to collect data
+  const enabled = await getSetting("analytics").getValue();
```

Comment:
You should use the `getSetting().getValue()` function for getting setting values.